### DISCUSSION TITLE
Change the markdown path for localisation (i18n)

### DISCUSSION THREAD

**Discussion by leoplct** [markdoc/markdoc#339] [Q&A] [answered]
We structure our localization in the same folder but with different filename.
**Example**
src/pages/docs/analytics.**en**.md (locale=en)
src/pages/docs/analytics.**fr**.md (locale=fr)
src/pages/docs/analytics.**es**.md (locale=es)

or alternately,

src/pages/docs/fr/analytics.md
src/pages/docs/en/analytics.md
src/pages/docs/es/analytics.md

Markdoc for nextjs always renders from filename.md ([documentation](https://markdoc.dev/docs/nextjs))
How can I say to render from filename.{locale}.md ?

where locale is from [Nextjs i18n](https://nextjs.org/docs/advanced-features/i18n-routing)

```
import { useRouter } from 'next/router'
let router = useRouter()
const { locale, defaultLocale, locales } = useRouter()
```

**Comment by mfix-stripe** [comment, marked as answer]
Hey @leoplct, sorry for the delay. You just need to follow the normal Next.js internationalized routing, which should support your use case: https://nextjs.org/docs/pages/building-your-application/routing/internationalization

Let me know if that doesn't work.

**Reply by cryptobench** [reply]
Hey @mfix-stripe !

We're trying to setup internationalization up in our already fantastic markdoc + nextjs project. We usually had the path

`/pages/docs/*` which contained all our .md files


Now we're looking into adding additional languages to our documentation, but we're struggling a bit. We've follow the next.js guide above and configured our next config as following:

```js
import withMarkdoc from '@markdoc/next.js'
import withSearch from './src/markdoc/search.mjs'

/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  pageExtensions: ['md', 'mdoc', 'js', 'jsx', 'ts', 'tsx'],
  experimental: {
    scrollRestoration: true,
  },
  distDir: 'dist',
  i18n: {
    locales: ['en', 'ja'],
    // This is the default locale you want to be used when visiting
    // a non-locale prefixed path e.g. `/hello`
    defaultLocale: 'en',
  },
}

// if (!process.env.VERCEL) {
//   nextConfig.output = 'export'
//   nextConfig.images = {
//     unoptimized: true,
//   }
// }

export default withSearch(
  withMarkdoc({ schemaPath: './src/markdoc' })(nextConfig)
)

````

And we've updated our pages structure to `/pages/en/docs/*` and `/pages/ja/docs/*` , but when trying to access the sites we receive a 404. I tried adding back our old /pages structure `/pages/docs/*` alongside the new ones, and I can see that Nextjs only tries to render via `pages/docs/*` and not `/pages/{detectedLocale}/docs`.

Do you have any clue to why this might be?

**Reply by cryptobench** [reply]
I see that when I remove the i18 settings, then the routing works, but when enabling it i get 404.
